src/index.js

```diff
@@ -1,74 +1,102 @@
-/* oauth login ? */
+var contact_us_el = document.getElementById('contact-us');
+var logo_el = document.getElementById('logo');
+var footer_eu_el = document.getElementById('footer-eu');
+var footer_non_eu_el = document.getElementById('footer-non-eu');
+
 var href = window.location.href;
+var default_app_id = 11;
+
 //Remove the '#' check later once the backend changes are released TODO
 var params_str = href.indexOf('#') != -1 ? href.split('#')[1] : href.split('?')[1];
-if (/acct1=/.test(href) && /token1=/.test(href)) {
-    var accts = (params_str.match(/acct\d=[\w\d]+/g) || []).map(function (val) { return val.split('=')[1] });
-    var tokens = (params_str.match(/token\d=[\w\d-]+/g) || []).map(function (val) { return val.split('=')[1] });
-    var currs = (params_str.match(/cur\d=[\w\d]+/g) || []).map(function (val) { return val.split('=')[1] });
-    var oauth = [];
-    for (var i = 0; i < accts.length; i++) {
-        var id = accts[i];
-        var token = tokens[i];
-        var curr = currs[i];
-        oauth.push({ id: id, token: token, currency: curr });
-    }
-    local_storage.set('oauth', oauth);
-    local_storage.set('oauth-login', { value: true }); /* used to fire oauth-login event in binary_websockets.js */
-}
-
 var lang = (params_str && params_str.match(/lang=[a-zA-Z]+/g) || []).map(function (val) { return val.split('=')[1] })[0] ||
     (local_storage.get('i18n') && local_storage.get('i18n').value) || 'en';
 
-set_language(lang);
-clear_url_querystring(href);
+contact_us_el.href = getBinaryUrl('contact.html');
+logo_el.href = getBinaryUrl('home.html');
 
-populate_language_dropdown();
+checkRedirectToken(params_str);
 
-var contact_us_el = document.getElementById('contact-us');
-var logo_el = document.getElementById('logo');
+setLanguage(lang);
+clearUrlQuerystring(href);
 
-contact_us_el.href = getBinaryUrl('contact.html');
-logo_el.href = getBinaryUrl('home.html');
+populateLanguageDropdown();
 
-if (local_storage.get('oauth') !== null) {
-    window.location.href = VERSION + 'main.html';
-} else {
-    $(function () {
-        $('body').css('display', 'block');
-        setTime();
-        setInterval(setTime, 1000);
-
-        var i18n_name = (window.local_storage.get('i18n') || { value: 'en' }).value;
-        $.getJSON(VERSION + 'i18n/' + i18n_name + '.json', function (data) {
-            setup_i18n_translation(data);
-            populate_footer();
-            if (lang === 'id') {
-                $('#footer').hide();
-            }
-        });
+setSelectedLanguage();
```

Comment:
Can we rename this? since the function does more than setting the selectedLanguage. Also I think it makes sense to move `populateLanguageDropdown` inside of this function?